test/benchmarks/netperf/launch.go

```diff
@@ -0,0 +1,430 @@
+ 2. Launch the worker pods
+ 3. Wait for the output csv data to show up in orchestrator pod logs
+*/
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"k8s.io/kubernetes/pkg/api"
+	restclient "k8s.io/kubernetes/pkg/client/restclient"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/client/unversioned/clientcmd"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/util/intstr"
+)
+
+const (
+	debugLog         = "output.txt"
+	testNamespace    = "netperf"
+	csvDataMarker    = "GENERATING CSV OUTPUT"
+	csvEndDataMarker = "END CSV DATA"
+	netperfImage     = "girishkalele/netperf-latest"
+
+	runUUID          = "latest"
+	orchestratorPort = 5202
+	iperf3Port       = 5201
+	netperfPort      = 12865
+)
+
+var (
+	iterations     int
+	hostnetworking bool
+	tag            string
+
+	everythingSelector api.ListOptions
+
+	primaryNode   api.Node
+	secondaryNode api.Node
+)
+
+func init() {
+	flag.BoolVar(&hostnetworking, "hostnetworking", false, "(boolean) Enable Host Networking Mode for PODs")
+	flag.IntVar(&iterations, "iterations", 1, "Number of iterations to run")
+	flag.StringVar(&tag, "tag", runUUID, "Storage Tag to use for this rug")
+	everythingSelector = api.ListOptions{LabelSelector: labels.Everything(), FieldSelector: fields.Everything()}
+}
+
+func setupClient() *client.Client {
+	// Read user's kube config and use the server and credentials from the current context
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+	kubeConfig, err := loadingRules.Load()
+	if err != nil {
+		fmt.Println("Failure loading kubeconfig credentials and server location")
+		return nil
+	}
+
+	context, ok := kubeConfig.Contexts[kubeConfig.CurrentContext]
+	if !ok {
+		fmt.Println("Failed to find default kubectl context")
+		return nil
+	}
+	authInfo, ok := kubeConfig.AuthInfos[fmt.Sprintf("%s-basic-auth", context.AuthInfo)]
+	if !ok {
+		fmt.Println("Failed to find the default auth info in the kubectl config file")
+		return nil
+	}
+
+	fmt.Printf("Setting up kubeConfig parameters for REST client User %s Password ***** Host: %s\n",
+		authInfo.Username,
+		kubeConfig.Clusters[kubeConfig.Contexts[kubeConfig.CurrentContext].Cluster].Server)
+
+	config := &restclient.Config{
+		Username: authInfo.Username,
+		Password: authInfo.Password,
+		Host:     kubeConfig.Clusters[kubeConfig.Contexts[kubeConfig.CurrentContext].Cluster].Server,
+		Insecure: true,
+	}
+	c, err := client.New(config)
+	if err != nil {
+		fmt.Println("Failed to connect to Kubernetes API server", err)
+		return nil
+	}
+	return c
+}
+
+// getMinions : Only return minion nodes
+func getMinionNodes(c *client.Client) *api.NodeList {
+	var nodesTmp *api.NodeList
+	var nodes api.NodeList
+	var err error
+	if nodesTmp, err = c.Nodes().List(everythingSelector); err != nil {
+		fmt.Println("Failed to fetch nodes", err)
+		return nil
+	}
+
+	for _, n := range nodesTmp.Items {
```

Comment:
you can search for schedulable nodes alone instaed of string matching the name, see function in e2e/utils (getschedulablenodes or something)